aep/0133.yaml

```diff
@@ -0,0 +1,148 @@
+          type: object
+          required: ['parameters']
+          properties:
+            parameters:
+              type: array
+              contains:
+                type: object
+                required: ['name']
+                properties:
+                  name:
+                    enum: ['id']
+
+  aep-133-param-types:
+    description: The id parameter should be a query parameter of type string.
+    severity: error
+    formats: ['oas2', 'oas3']
+    given:
+      - '#CreateOperation.parameters[?(@.name == "id")]'
+    then:
+      function: schema
+      functionOptions:
+        schema:
+          type: object
+          required: ['in', 'schema']
+          properties:
+            in:
+              enum: ['query']
+            schema:
+              type: object
+              required: ['type']
+              properties:
+                type:
+                  type: string
+                  enum: ['string']
+
+  aep-133-request-body:
+    description: A standard create method must accept the resource in the request body.
+    message: The request body is not an AEP resource.
+    severity: error
+    formats: ['oas3']
+    given: '#CreateOperation'
+    then:
+      function: schema
+      functionOptions:
+        schema:
+          type: object
+          required: ['requestBody']
+          properties:
+            requestBody:
+              type: object
+              required: ['content']
+              properties:
+                content:
+                  type: object
+                  additionalProperties:
+                    type: object
+                    required: ['schema']
+                    properties:
+                      schema:
+                        type: object
+                        required: ['x-aep-resource']
+
+  aep-133-required-params:
+    description: A create operation must not have any required parameters other than path parameters.
+    severity: error
+    formats: ['oas2', 'oas3']
+    given:
+      - '#CreateOperation.parameters[?(@.in != "path")]'
+      - '#CreateOperation^.parameters[?(@.in != "path")]' # parameters at path item level
+    then:
+      field: required
+      function: falsy
+
+  aep-133-response-body:
+    description: The success response for a create operation must be the created AEP resource.
+    message: The response body is not an AEP resource.
+    severity: error
+    formats: ['oas3']
+    given: '#CreateOperation'
+    then:
+      function: schema
+      functionOptions:
+        schema:
+          type: object
+          required: ['responses']
+          properties:
+            responses:
+              type: object
+              anyOf:
+                - required: ['200']
+                - required: ['201']
+              properties:
+                '200':
+                  $ref: '#/definitions/response'
+                '201':
+                  $ref: '#/definitions/response'
+          definitions:
+            response:
```

Comment:
nit: is it possible to re-use these definitions? I see a lot of rules using this matcher to determine if it's a proper resource or not.